### DISCUSSION TITLE
Rename "LockupLL" to "LockupLinear" and "LockupLT" to "LockupTranched"

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/v2-periphery#314] [Ideas]
It just dawned on me that there is a redundancy in the names of the following contracts (on the `staging` branch):

- `SablierV2MerkleLockupLL`
- `SablierV2MerkleLockupLT`

Some background on how we arrived at these names - the `MerkleLockup` contract used to be called `MerkleStreamer`. When "streamer" was part of the name, it made sense to append "LL" and "LT" to distinguish between the underlying distribution contracts. But now, it feels a bit redundant to me. How about we rename the contracts like so?

- `SablierV2MerkleLL`
- `SablierV2MerkleLT`

Or like so:

- `SablierV2MerkleLockupLinear`
- `SablierV2MerkleLockupTranched`

Cc @sablier-labs/solidity and @razgraf.

**Comment by razgraf** [comment]
Yup, it is repetitive. Flagged this a while back (see note (1)): https://github.com/sablier-labs/v2-periphery/discussions/288#discussioncomment-8566861

**Reply by PaulRBerg** [reply]
Great, so which option do you like more?

**Reply by razgraf** [reply]
The long form I guess (2nd option)? That would follow the same structure as the names we use in [v2-core](https://github.com/sablier-labs/v2-core/tree/staging/src) i.e. `SablierV2LockupTranched`.

**Reply by andreivladbrg** [reply]
Long form works for me

**Reply by smol-ninja** [reply]
I like short form more but lets go with the majority.

**Reply by PaulRBerg** [reply]
I also like the short form more, which makes this a deuce. Rationale for short-term:

- We actually want to differentiate from V2 Core to avoid accidental confusion, e.g., a user seeing `MerkleLockupLinear` on Etherscan and thinking that this is a Core contract
- LockupLinear/ LockupTranched are technical distribution models anyway - users don't know about them, they only know about the models shown in the UI
- We might create distribution models with three or more words in the future

**Reply by andreivladbrg** [reply]
I am neutral - I vote for both

**Comment by PaulRBerg** [comment]
Opened an issue: https://github.com/sablier-labs/v2-periphery/issues/319
